### PR TITLE
Refactor form-response/latest to use newer endpoints

### PR DESCRIPTION
**Pull request by paulfalgout**
Shortcut Story ID: [sc-57621]

Now we're using clinicians/me/form-responses/latests for a latest response by the user
and patients/../form-responses/submitted for the use cases where we need a response from a different context

## Release Notes

- **New Features**
  - Streamlined fetching of form responses to focus on user-specific submissions.

- **Bug Fixes**
  - Enhanced integration tests to accurately reflect expected behaviors and improve error handling for form submissions.

- **Documentation**
  - Updated test cases and API request structures to align with new endpoint specifications and naming conventions.

- **Chores**
  - Removed unnecessary constants and simplified filter logic in various components.

### CODE REVIEW

File: src/js/apps/forms/form/form_app.js [hunk 1]
```diff
@@ -66,11 +65,7 @@ export default App.extend({
       Radio.request('entities', 'fetch:forms:byAction', this.patientActionId),
       Radio.request('entities', 'fetch:actions:withResponses', this.patientActionId),
       Radio.request('entities', 'fetch:patients:model:byAction', this.patientActionId),
-      Radio.request('entities', 'fetch:formResponses:latest', {
-        action: this.patientActionId,
-        status: FORM_RESPONSE_STATUS.ANY,
-        editor: this.currentUser.id,
-      }),
+      Radio.request('entities', 'fetch:formResponses:byMe', { actionId: this.patientActionId }),
```
[the remark this answers is not in the document]
Comment (paulfalgout) [reply src/js/apps/forms/form/form_app.js:68]: it isn't used in this file though..

**Review by shadowhand — COMMENTED** [review]
That did clean things up, excellent!

File: src/js/entities-service/form-responses.js [hunk 2]
```diff
@@ -8,25 +7,32 @@ const Entity = BaseEntity.extend({
...
+    return this.fetchBy(url, { data }).then(response => response || new Model());
+  },
+  fetchByMe({ actionId, patientId, formId }) {
+    const filter = actionId ? { action: actionId } : { patient: patientId, form: formId };
+
+    return this.fetchOrEmpty('/api/clinicians/me/form-responses/latest', { filter });
+  },
+  fetchSubmitedByPatient({ patientId, actionId, flowId, formId, actionTags, submittedAt }) {
+    const filter = {
+      ...(actionId && { actions: actionId }),
```
Comment (paulfalgout) [inline src/js/entities-service/form-responses.js:28]: this pattern was a chatgpt recommendation.. I'm on the fence.. the other option looks pretty much like it was with the filter method... but this translates the keys at the same time which is nice.

Comment (shadowhand) [reply src/js/entities-service/form-responses.js:28]: This seems easy enough to modify in the future. Looks decent to me.

**Review by nmajor25 — APPROVED** [review]

**Pull request merged**